pkg/controller/installconfig/installconfig_controller.go

```diff
@@ -194,14 +206,30 @@ func (r *ReconcileInstallConfig) ensureCheSubscription(logger logr.Logger, ns st
 	return subCreated, err
 }
 
-func (r *ReconcileInstallConfig) StatusUpdate(logger logr.Logger, installConfig *v1alpha1.InstallConfig, statusUpdater func(installConfig *v1alpha1.InstallConfig, message string) error, msg string) (reconcile.Result, error) {
+func (r *ReconcileInstallConfig) ensureTektonSubscription(logger logr.Logger, ns string, installConfig *v1alpha1.InstallConfig) (bool, error) {
+	tektonSub := tekton.NewSubscription(ns)
+	subCreated := false
+	if err := controllerutil.SetControllerReference(installConfig, tektonSub, r.scheme); err != nil {
+		return subCreated, err
+	}
+	sub := &olmv1alpha1.Subscription{}
+	err := r.client.Get(context.TODO(), types.NamespacedName{Name: tektonSub.GetName(), Namespace: tektonSub.GetNamespace()}, sub)
```

Comment:
I prefer first client.Get() and if not_found then create new_subscription and set ctrl_ref rather doing this way.